### PR TITLE
fix_1122_SECURITY

### PR DESCRIPTION
**Pull request by clanmills**
See #1122

WIP.  Initial version of SECURITY.md based on: https://www.libssh.org/development/security-process/

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+## The security release process
+
+Security issues are reported as issues on https://github.com/exiv2/exiv2 with the label Security.
+
+1.) Someone should feel responsible and make sure that it's really a security
+    defect. If yes, we need one person who is responsible for this specific
+    issue beyond this point. This person coordinates the further actions and
+    might delegate tasks, of course.
+
+2.) Create a bug report with limited access to "libssh core developers" only!
```
Comment (piponazo) [inline SECURITY.md:20]: There is a mention here to **libssh**. I guess we need to adapt the paragraph to Exiv2

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+4.) Do a CVSSv3 calculation ([2]) and add it to the bug report and the advisory.
+
+5.) Ask for a CVE (usually Red Hat Product Security  secalert@redhat.com (see
+    [1] for GPG key etc.) does help out on this one) using the initial
+    advisory as evidence.
+    After receiving the CVE number, please add it as an alias to the bug report
+    advisory and to the summary.
+
+To:      Red Hat Product Security  secalert@redhat.com
+Subject: Another libssh CVE (Name of issue)
```
Comment (piponazo) [inline SECURITY.md:36]: Another mention to libssh

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+8.) Finish the advisory and attach it to the bug report as
+    advisory-$CVE.txt.  Ask for review.  (This must be after getting
+    the release date so the correct version numbers can be included).
+
+9.) ~10 days before the release date, confirm every patch and the CVE
+    text still has the correct reviews (due to additional feedback for
+    example).
+
+9a.) Add vendors to the CC list in the bug report.
```
Comment (piponazo) [inline SECURITY.md:88]: I am not sure if this would apply to exiv2

File: SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+9b.) Add the planned release date to the bug or the tracking bug if
+     that has been created.  This will be the first useful e-mail seen
+     by our vendors.
+10.) 3 days before the release date, send an e-mail to
```
Comment (piponazo) [inline SECURITY.md:93]: We do not have a mailing light, right? We could also remove this point

**Review by piponazo — COMMENTED** [review]
Thanks for working on this. I provided you some feedback

**Comment by clanmills** [comment]
I'm going to close this PR.  I have provided the file SECURITY.md directly into GitHub using the tools on the Security Tab.

I admire what is described here:  https://www.libssh.org/development/security-process/

Exiv2 does not have the resources to undertake a security policy of similar scale to that adopted by libssh.  Moreover, I don't believe Exiv2 warrants making the resource commitment described by libssh.

However, in studying this matter, I have made (and documented) provision in the version numbering scheme to enable Exiv2 to issue security releases.  That scheme has scope for every release of Exiv2 v0.27 and above.  No security release of Exiv2 has ever been made.

**Comment by clanmills** [comment]
@piponazo 

Thanks for your comments.  I began by cut'n'pasting the stuff from libssh into SECURITY.md on the 0.27-maintenance branch.  Having thought more about this, we don't have the resources to do undertake what's being discussed by libssh.

I expect Andreas Schneider will be a little disappointed in the weakness described in our SECURITY.md.  I would rather be weak and honest that to commit to something that cannot be achieved.

I am happy about having SECURITY.md in the repos.  We are giving users a definition of our capability.

I've use the "Security" tab in GitHub to define SECUITY.md which is on the 'master' branch.  You're welcome to edit it of course.

**Pull request closed without merging**